### PR TITLE
Spike mobile iam multifactor fix

### PR DESCRIPTION
**Pull request by jperk51**
## Description of change
Mobile IAM users should be multifactor as long as they are loa3. 

## Original issue(s)
N/A

## Things to know about this PR
Fix is required for future features of this API. Tested against existing test cases. 

Note: This is a work around for the existing PPIU access policy. The current policy checks if a user is both loa3 and is multifactor. We were able to confirm with Perry Vessels (Perry.Vessels@va.gov) that a user can't be loa3 without also being multifactor so the existing policy should probably be changed, but we didn’t want to change code for a product we don’t own.

### CODE REVIEW

**Comment by jjulian** [comment]
We should define "multifactor" - does this really just mean LOA3? Some users are LOA2 (DSLogon I think). Do we have any way of knowing if they truly have MFA set up from the introspection?

**Comment by jperk51** [comment]
> We should define "multifactor" - does this really just mean LOA3? Some users are LOA2 (DSLogon I think). Do we have any way of knowing if they truly have MFA set up from the introspection?

Agreed, @kreek is there any better way to check if a user is multifactor?

**Comment by kreek** [comment]
> Agreed, @kreek is there any better way to check if a user is multifactor?

@jperk51 below is a test user example response from IAM, I don't think there's anything in there that describes if they have multi factor enabled. Perry <Perry.Vessels@va.gov> or Prashant <Prashant.Mukadam@va.gov> with the IAM team would know. I think they can add fields that are missing as well.

```json
{
  "fediamtransaction_id": "5eWibth5T4By92LinZBrcRi+dMHCkde5bhzwwVeOmPI=",
  "fediam_is_delegate": "false",
  "fediam_birls_number": "796121200",
  "birthdate": "1970-08-12",
  "fediamss_issue_instant": "2020-08-05T21:48:38Z",
  "fediam_mviicn": "1008596379V859838",
  "fediam_street": "1700 University Boulevard",
  "fediamsecid": "0000028114",
  "client_id": "VAMobile",
  "fediam_country": "NOT_FOUND",
  "fediam_gender": "MALE",
  "exp": 1596667726,
  "code_challenge": "tDKCgVeM7b8X2Mw7ahEeSPPFxr7TGPc25IV5ex0PvHI",
  "fediam_street1": "1700 University Boulevard",
  "fediam_do_dedipn_id": "1005079124",
  "fediam_gc_id": "1008596379V859838^NI^200M^USVHA^P|796121200^PI^200BRLS^USVBA^A|0000028114^PN^200PROV^USDVA^A|1005079124^NI^200DOD^USDOD^A|32331150^PI^200CORP^USVBA^A|85c50aa76934460c8736f687a6a30546^PN^200VIDM^USDVA^A|2810777^PI^200CORP^USVBA^A|32324397^PI^200CORP^USVBA^A|19798466a4b143748e664482c6b6b81b^PN^200VIDM^USDVA^A|796121200^AN^200CORP^USVBA^",
  "active": true,
  "fediamauth_n_type": "IDME",
  "fediam_not_on_or_after": "2020-08-05T21:53:42Z",
  "aud": "VAMobile",
  "fediam_mcid": "WSSOE2008051748411450069042554",
  "fediamidsource": "ssoe",
  "fediam_vaafi_proof_authority": "FICAM",
  "phone_number": "(858)335-0190",
  "tokens_generated_by": "OAuth AZN Code Flow",
  "fediamissuer": "https://int.eauth.va.gov/isam/sps/saml20idp/saml20",
  "fediamproofing_auth": "FICAM",
  "fediam_authentication_method": "http://idmanagement.gov/ns/assurance/loa/3",
  "fediamam_eai_xattr_session_lifetime": "1596667722",
  "fediam_suffix": "NOT_FOUND",
  "sub": "0000028114",
  "fediam_mhvien": "NOT_FOUND",
  "fediam_authentication_instant": "2020-08-05T21:48:42Z",
  "token_type": "bearer",
  "fediam_common_name": "va.api.user+idme.008@gmail.com",
  "scope": "openid",
  "fediam_postal_code": "78665",
  "fediam_vaafi_csp_id": "200VIDM_19798466a4b143748e664482c6b6b81b",
  "fediam_pn_id": "796121200",
  "fediam_pn_id_type": "SSN",
  "iat": 1596664126,
  "email": "va.api.user+idme.008@gmail.com",
  "code_challenge_method": "S256",
  "given_name": "GREG",
  "middle_name": "A",
  "fediamassur_level": "3",
  "fediam_not_before": "2020-08-05T21:43:42Z",
  "fediam_prefix": "NOT_FOUND",
  "fediam_state": "TX",
  "fediam_city": "Round Rock",
  "fediam_pid": "32331150,2810777,32324397",
  "family_name": "ANDERSON",
  "username": "0000028114"
}
```

**Comment by jjulian** [comment]
MFA status would come from the iDP, nit sure IAM has access to that. What is the requirement that is causing us to need `multifactor` status in the api?

**Comment by jperk51** [comment]
> MFA status would come from the iDP, nit sure IAM has access to that. What is the requirement that is causing us to need `multifactor` status in the api?

It's one of the two checks that needed passed to confirm access to the PPIU service where this payment data comes from. 
`  def access?
    user.loa3? && user.multifactor
  end`
from ppiu_policy.rb in vets_api

File: app/models/iam_user_identity.rb [hunk 1]
```diff
@@ -38,6 +38,10 @@ def self.build_from_iam_profile(iam_profile)
     identity
   end
 
+  def multifactor
+    loa[:current].try(:to_i) == LOA::THREE
```
Comment (kreek) [inline app/models/iam_user_identity.rb:42]: think this can use ruby's [safe navigator](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/)

Comment (jperk51) [reply app/models/iam_user_identity.rb:42]: on it

**Review by kreek — APPROVED** [review]
LGTM 🚀

**Review by thilton-oddball — APPROVED** [review]
LGTM

**Pull request merged**